src/box/memtx_engine.cc

```diff
@@ -1774,7 +1780,15 @@ memtx_engine_schedule_gc(struct memtx_engine *memtx,
 			 struct memtx_gc_task *task)
 {
 	stailq_add_tail_entry(&memtx->gc_queue, task, link);
-	fiber_wakeup(memtx->gc_fiber);
+	if (memtx->gc_fiber == NULL) {
+#ifdef ENABLE_ASAN
+		bool stop = false;
+		while (!stop)
+			memtx_engine_run_gc(memtx, &stop);
```

Comment:
I think this should be done when the engine is destroyed, not when we schedule garbage collection.